include/boost/regex/v5/w32_regex_traits.hpp

```diff
@@ -62,7 +79,7 @@ namespace BOOST_REGEX_DETAIL_NS{
 //
 // start by typedeffing the types we'll need:
 //
-typedef std::uint32_t lcid_type;   // placeholder for LCID.
+typedef unsigned long lcid_type;        // placeholder for LCID.
```

Comment:
This won't work the same with Cygwin, where `long` is 64-bit. You can use `boost::uint32_t`. Or check for `__LP64__`, like you do below.

Reply:
Good catch, I might need to check the size of long or something, the issue is that this need to be *exactly* the same type as windows.h uses, otherwise there are errors when this header and windows.h are both included.  Same will be true with mingw/cygwin's flavour of windows.h.   I'll look into that shortly.

Reply:
`windows.h` is certainly present on Cygwin, and users could be including it on their own. Besides, that's an ABI discrepancy - I'm not sure what will be in the upper bits when `lcid_type` is returned from a WinAPI function. I still consider this a problem.

Reply:
>windows.h is certainly present on Cygwin, and users could be including it on their own. Besides, that's an ABI discrepancy - I'm not sure what will be in the upper bits when lcid_type is returned from a WinAPI function. I still consider this a problem.

My point is that w32_regex_traits.hpp is an empty header (everything is #ifdef'ed out) for anything except MSVC, so there should never be any clash because regex won't be defining those symbols on Cygwin/Mingwin.

Reply:
Oh, I see, you define `BOOST_REGEX_NO_W32` on Cygwin [here](https://github.com/boostorg/regex/blob/78bf0acd20c9c9fb96b7b53e46feb4880d1e9d35/include/boost/regex/config.hpp#L222).